paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java

```diff
@@ -84,10 +85,23 @@ default Optional<MetastoreClient.Factory> metastoreClientFactory(Identifier iden
     /**
      * Get the names of all databases in this catalog.
      *
-     * @return a list of the names of all databases
+     * @return a list of the names of user databases
      */
     List<String> listDatabases();
```

Comment:
It seems we should modify the implementation of `listDatabases`?

Hi @ulysses-you 
Could you research whether system database in database systems can be viewed with a list database command, such as in MySQL, PostgreSQL, etc.

Reply:
yea, I tried mysql and PostgreSQL, both of them will show the sys databases if user has permission.
```sql
mysql> show databases;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| mysql              |
| performance_schema |
| sys                |
+--------------------+
```

```sql
postgres=# \l
                                            数据库列表
   名称    | 拥有者 | 字元编码 | 校对规则 | Ctype | ICU Locale | Locale Provider |    存取权限
-----------+--------+----------+----------+-------+------------+-----------------+-----------------
 postgres  | cathy  | UTF8     | C        | C     |            | libc            |
 template0 | cathy  | UTF8     | C        | C     |            | libc            | =c/cathy       +
           |        |          |          |       |            |                 | cathy=CTc/cathy
 template1 | cathy  | UTF8     | C        | C     |            | libc            | =c/cathy       +
           |        |          |          |       |            |                 | cathy=CTc/cathy
(3 行记录)

```

Reply:
Thanks @ulysses-you 

We should implement `listDatabases` in `AbstractCatalog`.

Reply:
@JingsongLi  How about change ` List<String> listDatabases();` to ` List<String> listDatabases(boolean includeSystem);`? I'm afraid always return sys database will break somethings. I beleive the usage of `listDatabases` Paimon internal do not need sys database.